### PR TITLE
Improve the `cols_merge_uncert()` function

### PR DESCRIPTION
**Pull request by rich-iannone**
This change adds the `sep` argument to the `cols_merge_uncert()` function, making its interface consistent with that of `cols_merge_range()`. This change also uses an existing underlying pattern to provide the correct mark for the output context. Finally, this fixes a bug that appears in rare circumstances (with table output to IE11, using **gt** with R 3.5.3).

Fixes: https://github.com/rstudio/gt/issues/476.

### CODE REVIEW

File: R/modify_columns.R [hunk 1]
```diff
@@ -839,14 +840,13 @@ cols_hide <- function(data,
 cols_merge_uncert <- function(data,
                               col_val,
                               col_uncert,
+                              sep = " +/- ",
```
Comment (jcheng5) [inline R/modify_columns.R]: Allow `I(" +/- ")` to mean "use this literal value"

File: R/utils.R [hunk 2]
```diff
@@ -810,12 +810,32 @@ split_scientific_notn <- function(x_str) {
 #' @noRd
 tidy_gsub <- function(x, pattern, replacement, fixed = FALSE) {
 
-  gsub(pattern, replacement, x, fixed = fixed)
+  if (!utf8_aware_sub) {
```
Comment (jcheng5) [inline R/utils.R]: Please add unit tests

```r
identical(tidy_sub(".", ".", "\u00B1", fixed = TRUE), "\u00B1")
identical(tidy_gsub(".", ".", "\u00B1", fixed = TRUE), "\u00B1")
```

File: R/utils_formatters.R [hunk 3]
```diff
@@ -306,7 +306,7 @@ context_missing_text <- function(missing_text,
                missing_text
```
Comment (jcheng5) [inline R/utils_formatters.R]: It'd be nice if these patterns also respected `I()`

File: R/modify_columns.R (hunk 1 above)
Comment (rich-iannone) [reply R/modify_columns.R]: Now allowed and explained in docs.

File: R/utils.R (hunk 2 above)
Comment (rich-iannone) [reply R/utils.R]: These tests were added (and tests for usage of `I()` were added as well).

**Pull request merged**